src/plugins/data_source/server/routes/test_connection.ts

```diff
@@ -30,18 +30,19 @@ export const registerTestConnectionRoute = (
                   schema.literal(AuthType.NoAuth),
                   schema.literal(AuthType.SigV4),
                 ]),
-                credentials: schema.oneOf([
-                  schema.object({
-                    username: schema.string(),
-                    password: schema.string(),
-                  }),
-                  schema.object({
-                    region: schema.string(),
-                    accessKey: schema.string(),
-                    secretKey: schema.string(),
-                  }),
-                  schema.literal(null),
-                ]),
+                credentials: schema.maybe(
```

Comment:
credentials are optional because 1. testing existing datasource doesn't require passing credentials 2. testing `no_auth` type datasource doesn't require passing credential